runtime/near-vm-runner/src/logic/gas_counter.rs

```diff
@@ -116,15 +116,18 @@ impl GasCounter {
         }
     }
 
-    /// Simpler version of `deduct_gas()` for when no promises are involved.
+    /// Checks whether the current contract execution is allowed to burn this much new gas.
```

Comment:
nit: as far as I understand gas counter changes are no longer required, maybe consider reverting this